org.abapgit.adt.ui/src/org/abapgit/adt/ui/internal/util/GitCredentialsService.java

```diff
@@ -0,0 +1,355 @@
+					//store the credentials in secure storage
+					storeCredentialsInSecureStorage(repositoryCredentials, repositoryURL);
+				}
+			}
+			else {
+				repositoryCredentials = null;
+			}
+		});
+		return repositoryCredentials;
+	}
+
+	/**
+	 * Opens a dialog for the user to enter the credentials for the git account to access the given repository URL
+	 */
+	public static IExternalRepositoryInfoRequest getCredentialsFromUser(Shell shell, String repositoryURL) {
+		return getCredentialsFromUser(shell, repositoryURL, null);
+	}
+
+	/**
+	 * Checks whether the exception is of type authentication error
+	 *
+	 * @param httpStatus
+	 *            Http status code from the exception
+	 * @return true if the error code is of type authentication issue
+	 */
+	public static boolean isAuthenticationIssue(ResourceException exception) {
+		if (exception.getExceptionProperties() != null) {
+			String httpStatusValue = exception.getExceptionProperties().get(keyhttpStatus);
+			if (httpStatusValue != null && httpStatusValue.length() > 0) {
+				int httpStatus = Integer.parseInt(httpStatusValue);
+				/*
+				 * Authentication issues will result in error codes 401, 403 and 404
+				 * Check the following documentation from Guthub :
+				 * "There are two ways to authenticate through GitHub API v3. Requests that require authentication will return  404 Not Found ,
+				 *  instead of  403 Forbidden , in some places. This is to prevent the accidental leakage of private repositories to unauthorized users."
+				 */
+				if (httpStatus == 401 || httpStatus == 403 || httpStatus == 404) {
+					return true;
+				}
+			}
+		}
+		return false;
+	}
+
+	/**
+	 *
+	 * @param repositoryCredentials
+	 * @param repositoryURL
+	 *            store repositoryCredentials in Secure Store for the given
+	 *            repositoryURL
+	 */
+	public static void storeCredentialsInSecureStorage(IExternalRepositoryInfoRequest repositoryCredentials, String repositoryURL) {
+		if (repositoryCredentials != null && repositoryURL != null) {
+			String hashedURL = getMD5HashForUrl(repositoryURL);
+
+			//Disable security questions
+			Map<String, Boolean> options = new HashMap<String, Boolean>();
+			options.put(IProviderHints.PROMPT_USER, false);
+
+			ISecurePreferences preferences = null;
+			try {
+				preferences = SecurePreferencesFactory.open(null, options);
+			} catch (IOException e) {
+				AbapGitUIPlugin.getDefault().getLog().log(new Status(IStatus.ERROR, AbapGitUIPlugin.PLUGIN_ID, e.getMessage(), e));
+			}
+
+			if (preferences != null && hashedURL != null) {
+				ISecurePreferences node = preferences.node(hashedURL);
+				try {
+					node.put("user", repositoryCredentials.getUser(), false); //$NON-NLS-1$
+					node.put("password", repositoryCredentials.getPassword(), true); //$NON-NLS-1$
+				} catch (StorageException e) {
+					AbapGitUIPlugin.getDefault().getLog().log(new Status(IStatus.ERROR, AbapGitUIPlugin.PLUGIN_ID, e.getMessage(), e));
+			}
```

Comment:
[SUGGESTION] You can probably move this code under the try catch which is added before.

Reply:
Made the suggested changes.